### PR TITLE
Check if tcpdump binary exists

### PR DESCRIPTION
**Pull request by guedou**
This PR fixes issues discovered while investigating #1929.

### CODE REVIEW

File: scapy/sendrecv.py [hunk 1]
```diff
@@ -844,6 +845,11 @@ def sniff(count=0, store=True, offline=None, prn=None, lfilter=None,
             sniff_sockets[opened_socket] = "socket0"
     if offline is not None:
         flt = karg.get('filter')
+
+        if not TCPDUMP and not flt is None:
+            message = "tcpdump is not available. Cannot use filter!"
+            raise Scapy_Exception(message)
```
Comment (gpotter2) [inline scapy/sendrecv.py]: Do you think we should try to fallback on `conf.use_pcap` when tcpdump isn’t available ?

**Comment by guedou** [comment]
I do not think that we use libpcap anywhere else to compile filters. I have not plan to experiment with that =)

File: scapy/sendrecv.py (hunk 1 above)
Comment (gpotter2) [reply scapy/sendrecv.py]: Also: you don't need tcpdump with `conf.pcap` set to `True`. That's why it's failing on windows, where it won't use tcpdump anyways

File: scapy/utils.py [hunk 2]
```diff
@@ -1437,6 +1438,9 @@ def tcpdump(pktlist, dump=False, getfd=False, args=None,
         prog = [conf.prog.tcpdump]
     elif isinstance(prog, six.string_types):
         prog = [prog]
+    if not WINDOWS and not _check_tcpdump():
```
Comment (gpotter2) [inline scapy/utils.py]: Also also, it's overkill to call `_check_tcpdump` (
```python
with open(os.devnull, 'wb') as devnull:
    try:
        proc = subprocess.Popen([conf.prog.tcpdump, "--version"],
                                stdout=devnull, stderr=subprocess.STDOUT)
     except OSError:
        return False
```
) on each opened socket. There should be a cache or something

**Comment by guedou** [comment]
I broke this PR with a bad rebase =/

**Comment by guedou** [comment]
@gpotter2 if you have a local copy, could you create another PR?

**Comment by gpotter2** [comment]
@guedou I did have a copy. I've restored this PR and rebased it against master

**Comment by guedou** [comment]
@gpotter2 thanks! It looks good to me. I restarted the failing Travis test. You can merge the PR when it pass.

**Pull request merged**